sources/packages/backend/apps/api/src/route-controllers/program-year/program-year.student.controller.ts

```diff
@@ -6,25 +6,33 @@ import {
   ParseIntPipe,
 } from "@nestjs/common";
 import { ProgramYearService } from "../../services";
-import { OptionItem } from "../../types";
+import { ClientTypeBaseRoute } from "../../types";
 import BaseController from "../BaseController";
 import { AllowAuthorizedParty } from "../../auth/decorators/authorized-party.decorator";
 import { AuthorizedParties } from "../../auth/authorized-parties.enum";
-import { ProgramYearDto } from "./models/program-year.dto";
-import { ApiTags } from "@nestjs/swagger";
+import { ProgramYearAPIOutDTO } from "./models/program-year.dto";
+import { ApiNotFoundResponse, ApiTags } from "@nestjs/swagger";
 import { RequiresStudentAccount } from "../../auth/decorators";
+import { OptionItemAPIOutDTO } from "../models/common.dto";
 
 @AllowAuthorizedParty(AuthorizedParties.student)
 @RequiresStudentAccount()
 @Controller("program-year")
-@ApiTags("program-year")
+@ApiTags(`${ClientTypeBaseRoute.Student}-program-year`)
 export class ProgramYearController extends BaseController {
   constructor(private readonly programYearService: ProgramYearService) {
     super();
   }
 
+  /**
+   * Gets a list of program years returned as option items (id/description pair).
+   * @returns an array of program years as OptionItemAPIOutDTO.
+   */
   @Get("/options-list")
-  async getProgramYears(): Promise<OptionItem[]> {
+  @ApiNotFoundResponse({
+    description: "No program years were found.",
+  })
+  async getProgramYears(): Promise<OptionItemAPIOutDTO[]> {
```

Comment:
I know it was `OptionItem` earlier, just a suggestion, it would be better, if we could give a proper name, something like `ProgramYearOptionsAPIOutDTO`

Reply:
At first I thought of creating `ProgramYearOptionsAPIOutDTO` but then I see this `OptionItemAPIOutDTO` that is generic and used for other controllers.